homeassistant/components/asuswrt/config_flow.py

```diff
@@ -101,132 +128,214 @@ def _get_ip(host: str) -> str | None:
-            if result == RESULT_SUCCESS:
-                if unique_id:
-                    await self.async_set_unique_id(unique_id)
-                # we allow configure a single instance without unique id
-                elif self._async_current_entries():
-                    return self.async_abort(reason="invalid_unique_id")
-                else:
-                    _LOGGER.warning(
-                        "This device does not provide a valid Unique ID."
-                        " Configuration of multiple instance will not be possible"
-                    )
+            return self._show_setup_form(error="pwd_or_ssh")
+        if pwd and ssh:
+            return self._show_setup_form(error="pwd_and_ssh")
+        if ssh:
+            if not await self.hass.async_add_executor_job(_is_file, ssh):
+                return self._show_setup_form(error="ssh_not_file")
 
-                return self.async_create_entry(
-                    title=host,
-                    data=user_input,
-                )
+        host: str = user_input[CONF_HOST]
+        if not await self.hass.async_add_executor_job(_get_ip, host):
+            return self._show_setup_form(error="invalid_host")
+
+        result, unique_id, protocol = await self._async_check_connection(user_input)
+        if result == RESULT_SUCCESS:
+            self._unique_id = unique_id
+            self._config_data = {**user_input, CONF_PROTOCOL: protocol}
+            if protocol in [PROTOCOL_SSH, PROTOCOL_TELNET]:
+                return await self.async_step_legacy()
+            return await self._async_save_entry()
+
+        return self._show_setup_form(error=result)
+
+    async def async_step_legacy(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow for legacy settings."""
+        if user_input is None:
+            return self.async_show_form(step_id="legacy", data_schema=LEGACY_SCHEMA)
 
-            errors["base"] = result
+        self._config_data.update(user_input)
+        return await self._async_save_entry()
 
-        return self._show_setup_form(user_input, errors)
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._config_data[CONF_HOST] = entry_data[CONF_HOST]
+        self._config_data[CONF_USERNAME] = entry_data[CONF_USERNAME]
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
+            )
```

Comment:
This will bring up a blank dialog with the Submit button. I think here we should ask the user for the password (I'm not sure about the username). Please take a look [here](https://github.com/home-assistant/core/blob/84979f8e920889ed8974f215c4252c39c40b7389/homeassistant/components/shelly/config_flow.py#L311) for example.
```suggestion
            return self.async_show_form(
                step_id="reauth_confirm",
                data_schema=vol.Schema(
                    {vol.Required(CONF_PASSWORD): str}
                ),
            )
```

Reply:
My error, I misunderstand the explanation in documentation. I review the logic now: it will try to authenticate with new password using only http and https, if this fails it restore back original configuration (for old router that do not support http). The failure to migrate to http result in a different `abort` message